Satellite registration followup

avitova opened the pull request:
Yooo! So this is the Satellite followup. Basically, after inserting the following Satellite command and Certificate:

<details>
  <summary>Satellite command</summary>
    ```
set -o pipefail && curl -sS ‘https://ec2-107-23-89.compute-1.amazonaws.com/register?activation_keys=ak&location_id=2&organization_id=1&update_packages=false’ -H ‘Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjoiZGVtb1VzZXIiLCJleHAiOjMwMDAwMDAwMDB9.XYZ123ABC456DEF789GHI012JKL345MNO678PQR901 | bash
  ```
</details>

<details>
  <summary>Certificate</summary>
    ```
-----BEGIN CERTIFICATE-----
MIIDUDCCAjigAwIBAgIUIE7ftn9J9krO6TRJg8M3plAZGgEwDQYJKoZIhvcNAQEL
BQAwYjELMAkGA1UEBhMCVVMxEzARBgNVBAgMCkNhbGlmb3JuaWExFjAUBgNVBAcM
DVNhbiBGcmFuY2lzY28xETAPBgNVBAoMCFRlc3QgT3JnMRMwEQYDVQQDDAp0ZXN0
LmxvY2FsMB4XDTI1MDIwNTEzMzk0N1oXDTI2MDIwNTEzMzk0N1owYjELMAkGA1UE
BhMCVVMxEzARBgNVBAgMCkNhbGlmb3JuaWExFjAUBgNVBAcMDVNhbiBGcmFuY2lz
Y28xETAPBgNVBAoMCFRlc3QgT3JnMRMwEQYDVQQDDAp0ZXN0LmxvY2FsMIIBIjAN
BgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy8JdVoIaWh0PO1dL7xuGUNBUx6hZ
PBYSnWpPS7lNL3Y/KHNNZhStm0ISFYcB4C/mlJN+9kMcl3CXoktZHfrkencRwhlv
9aua70fZmmjgHDn3Stm25pqrhehUzoKZPlai9eXGJfY1q52ZMjNa0dxJjt6IST8U
oAwwXrBr14dUjuMM0ZhLeLtiTAh1Eb8CnXMmVmkhoMBbMODE3Lkqr72K8kseu8Qx
6Iq96ggkwiAQr3+h2GOkqtEl6BQEbjG1CVlVMCTU3B3yJ/uDUYvqK3897PdgWkUQ
2L3dZPTWv+p8+UjaC4zVYGnM7NpJisMZPXsbA9KiqaF+bUvLLjP9budPXwIDAQAB
MA0GCSqGSIb3DQEBCwUAA4IBAQCz2uByr3Tf34zSeYhy1z6MLJR4ijcfuWhxuE7D
M5fB4I0Ua3K6+ptZDvlWuikF+5InnoU3HSfrXVPCJ1my4jsgk+c4YKPW0yVRrr6m
hS2CKZyngICWnGCIYrlXlKeNJe4j23WF7IRhsykvkpt69Vw1x99UIJBcobOx+Kw/
zB92/XFIBwOZArUmGDaiL5MnqhmFWfc6mtIELxIRKCj9LQG9y7L1JoVyqug3thgZ
CdoLGtbHXri9BSR+8ogXu4JWp0YwMHTul6AEb2kcSZHTrYj6lUkXJMsw+E5jV37G
jZKGigLMSUp2z4jT+aX+HblYHrvTbrKct23EMeJeANQzF08e
-----END CERTIFICATE-----
    ```
</details>

...we have the following request:



<details>
  <summary>Request</summary>

  ```
{
  "name": "rhel-9-x86_64-04032025-1603",
  "description": "",
  "distribution": "rhel-9",
  "image_requests": [
    {
      "architecture": "x86_64",
      "image_type": "guest-image",
      "upload_request": {
        "type": "aws.s3",
        "options": {}
      }
    }
  ],
  "customizations": {
    "files": [
      {
        "path": "/etc/systemd/system/register-satellite.service",
        "data": "W1VuaXRdCiAgRGVzY3JpcHRpb249Q3VzdG9tIGZpcnN0IGJvb3Qgc2NyaXB0CiAgQ29uZGl0aW9uRmlsZUlzRXhlY3V0YWJsZT0vdXNyL2xvY2FsL3NiaW4vcmVnaXN0ZXItc2F0ZWxsaXRlCiAgQ29uZGl0aW9uUGF0aEV4aXN0cz0hL3Zhci9sb2NhbC8ucmVnaXN0ZXItc2F0ZWxsaXRlLWRvbmUKICBXYW50cz1uZXR3b3JrLW9ubGluZS50YXJnZXQKICBBZnRlcj1uZXR3b3JrLW9ubGluZS50YXJnZXQKICBBZnRlcj1vc2J1aWxkLWZpcnN0LWJvb3Quc2VydmljZQoKICBbU2VydmljZV0KICBUeXBlPW9uZXNob3QKICBFeGVjU3RhcnQ9L3Vzci9sb2NhbC9zYmluL3JlZ2lzdGVyLXNhdGVsbGl0ZQogIEV4ZWNTdGFydFBvc3Q9L3Vzci9iaW4vdG91Y2ggL3Zhci9sb2NhbC8ucmVnaXN0ZXItc2F0ZWxsaXRlLWRvbmUKICBSZW1haW5BZnRlckV4aXQ9eWVzCgogIFtJbnN0YWxsXQogIFdhbnRlZEJ5PWJhc2ljLnRhcmdldAogIFdhbnRlZEJ5PW11bHRpLXVzZXIudGFyZ2V0CiAgV2FudGVkQnk9Z3JhcGhpY2FsLnRhcmdldAo=",
        "data_encoding": "base64",
        "ensure_parents": true
      },
      {
        "path": "/usr/local/sbin/register-satellite",
        "data": "c2V0JTIwLW8lMjBwaXBlZmFpbCUyMCUyNiUyNiUyMGN1cmwlMjAtc1MlMjAlRTIlODAlOThodHRwcyUzQSUyRiUyRmVjMi0xMDctMjMtODkuY29tcHV0ZS0xLmFtYXpvbmF3cy5jb20lMkZyZWdpc3RlciUzRmFjdGl2YXRpb25fa2V5cyUzRGFrJTI2bG9jYXRpb25faWQlM0QyJTI2b3JnYW5pemF0aW9uX2lkJTNEMSUyNnVwZGF0ZV9wYWNrYWdlcyUzRGZhbHNlJUUyJTgwJTk5JTIwLUglMjAlRTIlODAlOThBdXRob3JpemF0aW9uJTNBJTIwQmVhcmVyJTIwZXlKaGJHY2lPaUpJVXpJMU5pSXNJblI1Y0NJNklrcFhWQ0o5LmV5SjFjMlZ5SWpvaVpHVnRiMVZ6WlhJaUxDSmxlSEFpT2pNd01EQXdNREF3TURCOS5YWVoxMjNBQkM0NTZERUY3ODlHSEkwMTJKS0wzNDVNTk82NzhQUVI5MDElMjAlN0MlMjBiYXNo",
        "data_encoding": "base64",
        "ensure_parents": true
      }
    ],
    "services": {
      "enabled": [
        "register-satellite"
      ]
    },
    "cacerts": {
      "pem_certs": [
        "-----BEGIN CERTIFICATE-----\nMIIDUDCCAjigAwIBAgIUIE7ftn9J9krO6TRJg8M3plAZGgEwDQYJKoZIhvcNAQEL\nBQAwYjELMAkGA1UEBhMCVVMxEzARBgNVBAgMCkNhbGlmb3JuaWExFjAUBgNVBAcM\nDVNhbiBGcmFuY2lzY28xETAPBgNVBAoMCFRlc3QgT3JnMRMwEQYDVQQDDAp0ZXN0\nLmxvY2FsMB4XDTI1MDIwNTEzMzk0N1oXDTI2MDIwNTEzMzk0N1owYjELMAkGA1UE\nBhMCVVMxEzARBgNVBAgMCkNhbGlmb3JuaWExFjAUBgNVBAcMDVNhbiBGcmFuY2lz\nY28xETAPBgNVBAoMCFRlc3QgT3JnMRMwEQYDVQQDDAp0ZXN0LmxvY2FsMIIBIjAN\nBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAy8JdVoIaWh0PO1dL7xuGUNBUx6hZ\nPBYSnWpPS7lNL3Y/KHNNZhStm0ISFYcB4C/mlJN+9kMcl3CXoktZHfrkencRwhlv\n9aua70fZmmjgHDn3Stm25pqrhehUzoKZPlai9eXGJfY1q52ZMjNa0dxJjt6IST8U\noAwwXrBr14dUjuMM0ZhLeLtiTAh1Eb8CnXMmVmkhoMBbMODE3Lkqr72K8kseu8Qx\n6Iq96ggkwiAQr3+h2GOkqtEl6BQEbjG1CVlVMCTU3B3yJ/uDUYvqK3897PdgWkUQ\n2L3dZPTWv+p8+UjaC4zVYGnM7NpJisMZPXsbA9KiqaF+bUvLLjP9budPXwIDAQAB\nMA0GCSqGSIb3DQEBCwUAA4IBAQCz2uByr3Tf34zSeYhy1z6MLJR4ijcfuWhxuE7D\nM5fB4I0Ua3K6+ptZDvlWuikF+5InnoU3HSfrXVPCJ1my4jsgk+c4YKPW0yVRrr6m\nhS2CKZyngICWnGCIYrlXlKeNJe4j23WF7IRhsykvkpt69Vw1x99UIJBcobOx+Kw/\nzB92/XFIBwOZArUmGDaiL5MnqhmFWfc6mtIELxIRKCj9LQG9y7L1JoVyqug3thgZ\nCdoLGtbHXri9BSR+8ogXu4JWp0YwMHTul6AEb2kcSZHTrYj6lUkXJMsw+E5jV37G\njZKGigLMSUp2z4jT+aX+HblYHrvTbrKct23EMeJeANQzF08e\n-----END CERTIFICATE-----\n"
      ]
    }
  }
}  ```
</details>

Make sure to break this, review this and comment.

regexowl commented on src/Components/CreateImageWizard/steps/Registration/SatelliteRegistration.tsx:
```diff
@@ -88,12 +87,12 @@ const SatelliteRegistration = () => {
                 : validated === 'error'
                 ? stepValidation.errors['certificate']
-                : 'Drag and drop a file or upload one'}
+                : 'Drag and drop a valid certificate file or upload one'}
```
Should a valid certificate validate as success in the helper text?

![image](https://github.com/user-attachments/assets/31f4aa9e-984d-4446-abad-fb4d8743131d)

regexowl reviewed (commented):
This looks and works great, thanks! Added some nitpicks.

Just out of curiosity - what are the plans around gating? I've noticed this is currently not available anywhere, will this PR push it over the threshold to get the registration to stage:preview?

regexowl commented:
Thanks for the fixes! A few more nitpicks, but other than those looks great!

command input doesn't show error for missing value
![image](https://github.com/user-attachments/assets/50c8b6bc-9672-4867-b44a-d10a9c15ebf9)

this validates as valid
![image](https://github.com/user-attachments/assets/5c8fd755-e9ff-4e13-ba37-c6c1a7f63660)

could you please add success variant to the condition for cert validation in the helper text?
![image](https://github.com/user-attachments/assets/c312f5dc-06df-43cb-a408-4d643df9fa87)

avitova commented:
> command input doesn't show error for missing value
Hmm, should we add the warning there? Since users have not added anything yet, I expect them to go and insert the command. I did not want to add a warning/error before something is done in that field if that makes sense. :thinking: I think it should be enough to flag that it is required. There is the token error so that we make sure that the registration command is valid.

> this validates as valid
Yeah. That is because I deleted the validation on these files. We want to support pem/der/cer, and there is not a sensible way to do that all on frontend. I think we should just limit the file extensions to these but not check the contents of the string. If you agree, I would leave that up to the users.

I added the success variant to the condition for cert validation. :)

regexowl commented:
> Hmm, should we add the warning there? Since users have not added anything yet, I expect them to go and insert the command. I did not want to add a warning/error before something is done in that field if that makes sense. 🤔 I think it should be enough to flag that it is required. There is the token error so that we make sure that the registration command is valid.

Ah yeah, the use case I had in mind, but forgot to mention (sorry about that :sweat_smile: ) is when user fills in the value and then removes it. We could probably render an error about required value based on pristinity of the field, to make the reason for disabled Next a bit more obvious. What do you think? I'm also perfectly fine with splitting that one into a follow up

> Yeah. That is because I deleted the validation on these files. We want to support pem/der/cer, and there is not a sensible way to do that all on frontend. I think we should just limit the file extensions to these but not check the contents of the string. If you agree, I would leave that up to the users.

Makes perfect sense

> I added the success variant to the condition for cert validation. :)

Thank you!

regexowl commented on package-lock.json:
These changes might have sneaked in somehow, I can't see any changes to `package.json`. Can you please drop the file? It should hopefully also resolve the current conflict :crossed_fingers:

avitova commented:
> Ah yeah, the use case I had in mind, but forgot to mention (sorry about that 😅 ) is when user fills in the value and then removes it. We could probably render an error about required value based on pristinity of the field, to make the reason for disabled Next a bit more obvious. What do you think? I'm also perfectly fine with splitting that one into a follow up

This is caused by the `  const [isPristine, setIsPristine] = useState(!value);` in [ValidatedInputAndTextArea](https://github.com/osbuild/image-builder-frontend/blob/463bfddae572aa18767f6b2bb76d05d0274b2660/src/Components/CreateImageWizard/ValidatedInput.tsx#L62), so I would not change the functionality in this pull request.

regexowl reviewed (approved):
Looks great, thank you :sparkles:
